azerothcore/azerothcore-wotlk
# fix(Core): Save bonus talent points to DB

## Changes Proposed:
-  Save player bonus talent points to the DB

## Issues Addressed:

- Allows bonus talents to be preserved through relogs
- Avoids forced talent reset on relog if bonus talent points had been granted
- Closes https://github.com/ZhengPeiRu21/mod-challenge-modes/issues/12

## SOURCE:


## Tests Performed:

- Tested in game, works


## How to Test the Changes:


1. Add bonus talents to player using SetBonusTalents (challenge modes module can be configured to do this)
2. Log out, confirm talents do not reset and bonus points are still available

## Known Issues and TODO List:



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
